### PR TITLE
Fix gradient caching + gradient symb comparison

### PR DESCRIPTION
**Pull request by markschlosseratbentley**
This fixes gradient texture caching on iTwin.js 3.0.  Previously, gradient textures would not be cached and the system would always create a new gradient texture even if the requested gradient texture matches a previous texture.

This also fixes gradient symb comparison (also a problem in pre-3.0 iTwin.js).  Previously, gradient symb comparison would not look at the underlying optional thematicSettings.  This means that changing certain thematic properties could result in the caching mechanism to think it should return an existing gradient texture, even if it really needs to make a new gradient texture.

See: https://github.com/iTwin/itwinjs-core/discussions/2818

Also, work items:
https://bentleycs.visualstudio.com/iModelTechnologies/_workitems/edit/765005
https://bentleycs.visualstudio.com/iModelTechnologies/_workitems/edit/765002

### CODE REVIEW

File: core/common/src/ThematicDisplay.ts [hunk 1]
```diff
@@ -119,6 +119,43 @@ export class ThematicGradientSettings {
     return true;
   }
 
+  /** Compares two sets of thematic gradient settings.
+   * @param lhs First set of thematic gradient settings to compare
+   * @param rhs Second set of thematic gradient settings to compare
+   * @returns 0 if lhs is equivalent to rhs, a negative number if lhs compares less than rhs, or a positive number if lhs compares greater than rhs.
+   */
+  public static compare(lhs: ThematicGradientSettings, rhs: ThematicGradientSettings): number {
+    if (lhs.mode < rhs.mode)
```
Comment (pmconne) [inline core/common/src/ThematicDisplay.ts:128]: Simplify.
```
let diff = 0;
if (diff = compareNumbers(lhs.mode, rhs.mode))
  return diff;
else if (diff = compareNumbers(lhs.stepCount, rhs.stepCount))
  return diff;
else if (diff = compareNumbers(lhs.marginColor.tbgr, rhs.marginColor.tbgr))
  return diff;
// etc...
```

Comment (markschlosseratbentley) [reply core/common/src/ThematicDisplay.ts:128]: Handy function.  Need to use compareWithTolerance(), though, am I right?  (Need to return -1, 0, or 1)

Comment (markschlosseratbentley) [reply core/common/src/ThematicDisplay.ts:128]: Oh, we just expect negative, positive or zero.  Will work out fine.  Need to change my test - I was expecting something different.  Sounds good.

Comment (markschlosseratbentley) [reply core/common/src/ThematicDisplay.ts]: Simplified it up.

**Review by pmconne — APPROVED** [review]

**Pull request merged**